### DISCUSSION TITLE
[Feature Request]: Support functions from Sqlite Extension Sqlean

### DISCUSSION THREAD

**Discussion by terefang** [duckdb/duckdb#14408] [Ideas]
### Why do you want this feature?

i have a number of use-cases that rely on sqlean extended functions and would also benefit from the functionality provided by duckdb.

as a matter of fact, duckdb already supports some of the extended functions from splean like uuid, md5 and sha256.

### what is missing from duckdb ?

* UUID related functions – https://github.com/nalgeon/sqlean/blob/main/docs/uuid.md
* Statistical Aggregates – https://github.com/nalgeon/sqlean/blob/main/docs/stats.md
* IP Manipulation – https://github.com/nalgeon/sqlean/blob/main/docs/ipaddr.md
    yes i know there is the INET extension but it is still missing functionality
* more Hash/Encoding Functions – https://github.com/nalgeon/sqlean/blob/main/docs/crypto.md

thanks in advance.

--
T

**Comment by terefang** [comment]
sorry for amending my request.

i have done futher research on extensions and found:

* ULID – that looks like a possible replacement for the uuid functions from sqlean
* crypto – that looks like a possible replacement for the hashing functions from sqlean.

sorry for the inconveniance.

**Comment by soerenwolfers** [comment]
duckdb offers a superset of the statistical functions in that extension, see https://duckdb.org/docs/sql/functions/aggregates.html#statistical-aggregates

**Reply by terefang** [reply]
> duckdb offers a superset of the statistical functions in that extension, see https://duckdb.org/docs/sql/functions/aggregates.html#statistical-aggregates

interesting ... looks like the quantiles have functional equivalence to the percentiles, which are flexible enough to fit my use-case.

that leaves IP-address manipulation/matching functions to be desired.
